src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
This block is a little bit complex to read because there are nested codes using ternary, it is recommended to use it with short statements like:
<br>
`const myValue = myCondition ? valueIfTrue : otherwise`


For complex statements is recommended to use the "if" to keep your code readability, for example in your code if you refactor for something like it:

```js
if (withChildren) {
  const keyForWomanSex = 'f'
  const womenWithChildren = people.filter(
    (p) => 
      p.sex === keyForWomanSex 
      && people.find((p1) => p1.mother === p.name)
  )

  ...your code here

  return ... your calculation
}

... your otherwise code 
```

you'll write more, but the readability is very important.

To help you refactor your code you can follow [your guidelines](https://mate-academy.github.io/style-guides/javascript.html#15-comparison-operators--equality) about comparison Operators & Equality with javascript.

Any question we're here to help you and explain some questions.